### ISSUE TITLE
Contradictory Errors When Configuring Unusual Boot Setups

### ISSUE THREAD

**Issue by useranon350** [NixOS/nixpkgs#316285]
### Describe the bug
Configuring ```boot.loader.generationsDir.enable = true;``` and ```boot.loader.systemd-boot.enable = true;``` causes an error due to multiple bootloaders, but disabling the latter causes an error due to ```You must set the option ‘boot.loader.grub.devices’ or 'boot.loader.grub.mirroredBoots' to make the system bootable.```

### Steps To Reproduce
1. Add ```boot.loader.generationsDir.enable = true;``` and ```boot.loader.systemd-boot.enable = true;``` to configuration.
2. Run nixos-rebuild boot
3. ...

### Screenshots
Not screenshots, but the errors.
### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by useranon350** [comment]
I think I found the issue. GRUB is the default bootloader, but is automatically disabled if systemd boot is enabled. Other boot options do not appear to have the same effect.

**Comment by Kamrat-Haggberg** [comment]
did you manage to fix it?

**Comment by useranon350** [comment]
Yes, I manually disabled GRUB using "boot.loader.grub.enable = false;"
